core/lib/prover_interface/src/inputs.rs

```diff
@@ -132,16 +135,50 @@ impl PrepareBasicCircuitsJob {
     }
 }
 
-/// Enriched `PrepareBasicCircuitsJob`. All the other fields are taken from the `l1_batches` table.
-#[derive(Debug, Clone)]
-pub struct BasicCircuitWitnessGeneratorInput {
-    pub block_number: L1BatchNumber,
-    pub previous_block_hash: H256,
-    pub previous_block_timestamp: u64,
-    pub block_timestamp: u64,
-    pub used_bytecodes_hashes: Vec<U256>,
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct VMRunWitnessInputData {
+    pub l1_batch_number: L1BatchNumber,
+    pub previous_root_hash: H256,
+    pub previous_meta_hash: H256,
+    pub previous_aux_hash: H256,
```

Comment:
I don't think that `previous_meta_hash` and `previous_aux_hash` should belong here. They are calculated by commitment generator asynchronously, so there are 2 ways:
- leave them here, but don't process batch in bwip until commitment generator finished the previous batch
- wait for these values in proof data handler
IMO second option is much better.